### PR TITLE
Disallow unquoted field names

### LINKED ISSUE #9800: Missing quotes breaks stuff
The following is allowed:
```
curl -XPUT localhost:9200/test9/test9/1 -d '{ test: 0 }'
```
However it upsets downstream systems as it's not quite JSON, eg. the Perl Search::Elasticsearch module, Kibana, and possibly others. Should the _source be rewritten to make it proper JSON (like when you submit YAML), or should this be banned entirely, maybe in 2.0?

### PR DESCRIPTION
**Pull request by jimmyjones2**
Closes #9800 - this change is breaking for 3.0 to disallow unquoted JSON which breaks a load of elasticsearch clients who expect Content-Type: application/json to be, well JSON :)

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContent.java [hunk 1]
```diff
@@ -43,7 +43,6 @@ public static XContentBuilder contentBuilder() throws IOException {
 
     static {
         jsonFactory = new JsonFactory();
-        jsonFactory.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, true);
```
Comment (dakrone) [inline core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContent.java]: I think we should explicitly set this to `false` so if Jackson's defaults ever change, we won't be bitten by it.

**Comment by dakrone** [comment]
Left one comment, other than that this looks good to me.

File: core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContent.java (hunk 1 above)
Comment (jimmyjones2) [inline core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContent.java]: Done!

Comment (dakrone) [inline core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContent.java]: Sweet, want to squash and I'll merge this?

Comment (dakrone) [inline core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContent.java]: Or I can just merge as-is, don't worry about it :)

**Comment by dakrone** [comment]
@jimmyjones2 actually, this causes failures in `DoSectionParserTests`, can you fix those tests before I merge this?

**Comment by jimmyjones2** [comment]
Humm, I wonder if this is breaking YAML... looks like a tomorrow job to me.

**Comment by clintongormley** [comment]
Hiya @jimmyjones2 

Did you manage to look into this any further?

**Comment by jimmyjones2** [comment]
@clintongormley Sorry completely forgot about this. A load of testcases were using unquoted JSON, have fixed them and rebased on master. Hope I haven't missed the boat for 5.0!

**Comment by dakrone** [comment]
Thanks @jimmyjones2 this looks great! I will run test and merge it

**Comment by dakrone** [comment]
Committed this to master (5.0), thanks again @jimmyjones2 !

**Pull request closed without merging**